platform.md

```diff
@@ -327,126 +388,75 @@ The lifecycle:
-| Output             | Description
-|--------------------|----------------------------------------------
-| [exit status]      | (see Exit Code table below for values)
-| `/dev/stdout`      | Logs (info)
-| `/dev/stderr`      | Logs (warnings, errors)
-| `<analyzed>`       | Analysis metadata (see [`analyzed.toml`](#analyzedtoml-toml)
-| `<layers>/<buidpack-id>/<layer>.sha`  | Files containing the diffID of each analyzed layer
+| Output                                | Description
+|---------------------------------------|----------------------------------------------
+| [exit status]                         | (see Exit Code table below for values)
+| `/dev/stdout`                         | Logs (info)
+| `/dev/stderr`                         | Logs (warnings, errors)
+| `<layers>/<buidpack-id>/store.toml`   | Persistent metadata (see data format in [Buildpack Interface Specification](buildpack.md))
 | `<layers>/<buidpack-id>/<layer>.toml` | Files containing the layer content metadata of each analyzed layer (see data format in [Buildpack Interface Specification](buildpack.md))
+| `<layers>/<buidpack-id>/<layer>/*`.   | Restored layer contents
 
 | Exit Code | Result|
 |-----------|-------|
 | `0`       | Success
 | `11`      | Platform API incompatibility error
 | `12`      | Buildpack API incompatibility error
 | `1-10`, `13-99` | Generic lifecycle errors
-| `200-299` | Analysis-specific lifecycle errors
+| `40-49` | Restoration-specific lifecycle errors
 
-- The lifecycle MUST write [analysis metadata](#analyzedtoml-toml) to `<analyzed>` if `<image>` is accessible.
-- **If** `<skip-layers>` is `true` the lifecycle MUST NOT perform layer analysis.
-- **Else** the lifecycle MUST analyze any app image layers or cached layers created by any buildpack present in the provided `<group>`.
+- For each buildpack in `<group>`, if persistent metadata for that buildpack exists in the analysis metadata, lifecycle MUST write a toml representation of the persistent metadata to `<layers>/<buildpack-id>/store.toml`
+- **If** `<skip-layers>` is `true` the lifecycle MUST NOT perform layer restoration.
+- **Else** the lifecycle MUST perform [layer restoration](#layer-restoration) for any app image layers or cached layers created by any buildpack present in the provided `<group>`.
 
-##### Layer analysis
-When analyzing a given layer the lifecycle SHALL:
+##### Layer Restoration
+When restoring a given layer the lifecycle SHALL:
 - **If** `build=true`, `cache=false`:
     - Do nothing
 - **Else if** `launch=true`:
     - Write layer metadata read from the analyzed image to `<layers>/<buildpack-id>/<layer-name>.toml`
     - Write the layer diffID from the analyzed image to `<layers>/<buildpack-id>/<layer-name>.sha`
+- **Else if** `cache=true` AND the cache DOES NOT contain a layer with matching diffID
+    - Must not write layer metadata
 - **Else if** `cache=true`:
     - Write layer metadata read from the cache to `<layers>/<buildpack-id>/<layer-name>.toml`
```

Comment:
Seems like we are missing a description of when to restore cached layer contents? In the interest of normalized independent API maybe we should just refer out to https://github.com/buildpacks/spec/blob/main/buildpack.md#layer-types for a description of what to restore when. Then, any future changes could be a buildpack API concern. We merely need to say that the lifecycle must use the provided `cache-dir` or `cache-image` to retrieve layer contents.

Reply:
I took a stab at removing all the detail here, but I might've removed too much. Let me know what you think. Fixed in 6480d2e975ba81d5b61d681f0177539408c834b8